week4/hw-week4.playground/Contents.swift

```diff
@@ -0,0 +1,165 @@
+    case .dog:
+        return ("A dog goes bow.")
+    case .elephant:
+        return ("A elephant goes trumpeting")
+    case .pig:
+        return ("A pig goes snort")
+    case .sheep:
+        return ("A sheep goes baa")
+    }
+}
+
+for animal in Animals.allCases{
+    print(theSoundMadeBy(animal: animal))
+}
+
+/*
+ c) This question will have you creating multiple functions that will require you to use closures and collections. First, you will do some setup.
+
+ Create an array of Int called nums with the values of 0 to 100.
+ Create an array of Int? called numsWithNil with the following values:
+ 79, nil, 80, nil, 90, nil, 100, 72
+
+ Create an array of Int called numsBy2 with values starting at 2 through 100, by 2.
+ Create an array of Int called numsBy4 with values starting at 2 through 100, by 4.
+
+ You can set the values of the arrays above using whatever method you find the easiest. In previous weeks you were introduced to ranges and sequences in Swift. Leveraging those in the Array initializer will allow you to create the requested arrays in a single line. Don’t let the last two break your stride.
+
+ - Create a function called evenNumbersArray that takes a parameter of [Int] (array of Int) and returns [Int]. The array of Int returned should contain all the even numbers in the array passed. Call the function passing the nums array and print the output.
+
+ - Create a function called sumOfArray that takes a parameter of [Int?] and returns an Int. The function should return the sum of the array values passed that are not nil. Call the function passing the numsWithNil array, and print out the results.
+
+ - Create a function called commonElementsSet that takes two parameters of [Int] and returns a Set<Int> (set of Int). The function will return a Set<Int> of the values in both arrays.
+
+ - Call the function commonElementsSet passing the arrays numsBy2, numsBy4, and print out the results.
+ */
+
+let nums = Array(stride(from: 0, through: 100, by: 1))
+let numsWithNil:[Int?] = [79, nil, 80, nil, 90, nil, 100, 72]
+let numsBy2 = Array(stride(from: 2, through: 100, by: 2))
+let numsBy4 = Array(stride(from: 2, through: 100, by: 4))
+
+/*
+ - Create a function called evenNumbersArray that takes a parameter of [Int] (array of Int) and returns [Int]. The array of Int returned should contain all the even numbers in the array passed. Call the function passing the nums array and print the output.
+*/
+
+func evenNumbersArray(val: [Int])->[Int]{
+    return val.filter {$0 % 2==0}
+}
+
+print(evenNumbersArray(val: nums))
+
+/*
+ - Create a function called sumOfArray that takes a parameter of [Int?] and returns an Int. The function should return the sum of the array values passed that are not nil. Call the function passing the numsWithNil array, and print out the results.
+ */
+
+func sumOfArray(val: [Int?])->Int{
+    let nonNilArray = val.compactMap{$0}
+    return nonNilArray.reduce(0,+)
+}
+
+print(sumOfArray(val: numsWithNil))
+
```

Comment:
I write my functions like this also, but you can get this to 1 line if you're interested

return val.compactMap{$0}.reduce(0, +)